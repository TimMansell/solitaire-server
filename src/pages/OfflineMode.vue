<template>
  <GameOverlay show-close data-test="offline-mode">
    <template #title> Offline mode</template>
    <template #msg>
      <p>
        An online database is used to keep track of game statistics. If, for
        some reason you become disconnected, you will be in offline mode.
      </p>
      <p>Offline mode happens when:</p>
      <ul>
        <li>The server goes offline</li>
        <li>Your connection to the server goes down or is interupted</li>
        <li>Your connection to the internet has gone down or is interupted</li>
        <li>The server is down for maintenance / updates</li>
      </ul>
      <p>How can I fix this?</p>
      <p>
        Usually the only thing you can do (other than waiting for server to come
        back online) is to refresh the page to initiate a new connection.
      </p>
    </template>
  </GameOverlay>
</template>

<script>
import GameOverlay from '@/components/GameOverlay.vue';

export default {
  name: 'OfflineMode',
  components: {
    GameOverlay,
  },
};
</script>
