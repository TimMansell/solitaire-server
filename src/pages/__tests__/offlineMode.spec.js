import { shallowMount } from '@vue/test-utils';
import OfflineMode from '@/pages/OfflineMode.vue';

describe('OfflineMode.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(OfflineMode);

    expect(wrapper).toMatchSnapshot();
  });
});
