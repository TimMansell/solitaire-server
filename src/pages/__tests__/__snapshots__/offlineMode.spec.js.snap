// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineMode.vue matches snapshot 1`] = `
<gameoverlay-stub showclose="true" data-test="offline-mode"> Offline mode<p>
    An online database is used to keep track of game statistics. If, for
    some reason you become disconnected, you will be in offline mode.
  </p>
  <p>Offline mode happens when:</p>
  <ul>
    <li>The server goes offline</li>
    <li>Your connection to the server goes down or is interupted</li>
    <li>Your connection to the internet has gone down or is interupted</li>
    <li>The server is down for maintenance / updates</li>
  </ul>
  <p>How can I fix this?</p>
  <p>
    Usually the only thing you can do (other than waiting for server to come
    back online) is to refresh the page to initiate a new connection.
  </p>
</gameoverlay-stub>
`;
